libraries/koinos_api_cpp/include/koinos/system/system_calls.hpp

```diff
@@ -1,128 +1,151 @@
+   detail::max_hash_size,
+   detail::max_argument_size >;
+
+using set_system_call_operation = koinos::protocol::set_system_call_operation< detail::max_argument_size >;
+using head_info = koinos::chain::head_info< detail::max_hash_size, detail::max_hash_size >;
 
-void apply_transaction( const protocol::transaction& t );
+void print( const std::string& s );
 
-void apply_reserved_operation( const protocol::reserved_operation& o );
+bool verify_block_signature( const std::string& digest, const std::string& active_data, const std::string& sig );
 
-void apply_upload_contract_operation( const protocol::create_system_contract_operation& o );
+void apply_block( const block& b, bool check_passive_data, bool check_block_signature, bool check_transaction_signatures );
 
-void apply_execute_contract_operation( const protocol::call_contract_operation& op );
+void apply_transaction( const transaction& t );
 
-void apply_set_system_call_operation( const protocol::set_system_call_operation& op );
+void apply_upload_contract_operation( const upload_contract_operation& o );
 
-bool db_put_object( const uint256& space, const uint256& key, const variable_blob& obj );
+void apply_execute_contract_operation( const call_contract_operation& op );
 
-template< typename T >
-bool db_put_object( const uint256& space, const uint256& key, const T& value )
-{
-   return db_put_object( space, key, pack::to_variable_blob( value ) );
-}
+void apply_set_system_call_operation( const set_system_call_operation& op );
 
-template< typename T >
-bool db_put_object( const uint256& space, variable_blob vb_key, const T& value )
-{
-   vb_key.resize( 32, char(0) );
-   auto key = pack::from_variable_blob< uint256 >( vb_key );
-
-   return system::db_put_object( space, key, pack::to_variable_blob( value ) );
-}
-
-variable_blob db_get_object( const uint256& space, const uint256& key, int32_t object_size_hint = -1 );
+bool put_object( const std::string& space, const std::string& key, const std::string& obj );
 
 template< typename T >
-bool db_get_object( const uint256& space, const uint256& key, T& t )
+bool put_object( const std::string& space, const std::string& key, const T& value )
 {
-   auto vb = db_get_object( space, key, sizeof( T ) );
-   if( vb.size() )
-   {
-      t = pack::from_variable_blob< T >( vb );
-      return true;
-   }
-
-   return false;
+   std::string obj;
+   value.SerializeToString( &obj );
+   return put_object( space, key, obj );
 }
 
+std::string get_object( const std::string& space, const std::string& key, int32_t object_size_hint = -1 );
+
 template< typename T >
-bool db_get_object( const uint256& space, variable_blob vb_key, T& t )
+bool get_object( const std::string& space, const std::string& key, T& t )
 {
-   vb_key.resize( 32, char(0) );
-   auto key = pack::from_variable_blob< uint256 >( vb_key );
-
-   auto vb = db_get_object( space, key, sizeof( T ) );
-   if( vb.size() )
+   auto obj = get_object( space, key, sizeof( T ) );
```

Comment:
I don't think we want to do `sizeof( T )` here. The size of the serialized fields is not the size of the struct. It likely is larger (although not always). The hint specifies the size of the buffer. If the hint is smaller than the size of the object, then the call will fail. We may want to change the semantics of this in chain, but that is the way it works right now.